lang/java/avro/src/main/java/org/apache/avro/reflect/ReflectData.java

```diff
@@ -796,20 +812,25 @@ public Protocol getProtocol(Class iface) {
   }
 
   private Message getMessage(Method method, Protocol protocol, Map<String, Schema> names,
-      Map<? extends Type, Type> genericTypeMap) {
+      Map<TypeVariable<?>, Type> genericTypeMap) {
     List<Schema.Field> fields = new ArrayList<>();
     for (Parameter parameter : method.getParameters()) {
-      Schema paramSchema = getSchema(genericTypeMap.getOrDefault(parameter.getParameterizedType(), parameter.getType()),
+      Schema paramSchema = getSchema(
+          genericTypeMap.getOrDefault(parameter.getParameterizedType(),
+              parameter.getParameterizedType() != null ? parameter.getParameterizedType() : parameter.getType()),
           names);
+      String paramName = parameter.getName();
       for (Annotation annotation : parameter.getAnnotations()) {
         if (annotation instanceof AvroSchema) // explicit schema
           paramSchema = new Schema.Parser().parse(((AvroSchema) annotation).value());
         else if (annotation instanceof Union) // union
           paramSchema = getAnnotatedUnion(((Union) annotation), names);
         else if (annotation instanceof Nullable) // nullable
           paramSchema = makeNullable(paramSchema);
+        else if (annotation instanceof AvroName)
```

Comment:
is this the part thats required for specifying `AvroName` annotation in method parameters in ipc? Is it possible to just use `parameter.getName()` instead?

Reply:
The parameter.getName() returns arg0. They removed dependency on the paranamer library which was used to generate paramater names in the protocol which has broken compatibility. Allowing to specify the name explicitly will allow us to work around the issue for existing protocols.